docs/design-proposals/node/Partial-Runtime-Readiness-Handling.md

```diff
@@ -9,20 +9,40 @@ ready or notReady. As a result, the node status and node readiness management ne
 a node with desired runtime service.
 
+Initially Arktos is currently implemented as "NodeReady requires ANY one of its runtime services ready" for best resource 
+utilization, i.e. the scheduler can schedule the pod to a node as long as its desired runtime service ready on the node.
+However, this implementation carries a drawback with PODs with cross workload type dependencies which can be described
+as followings:
+
+A cluster is setup and supports two runtime services, containerD and Arktos VM runtime,  and their corresponding workload
+types, i.e. Container workload type and VM workload type. The network agent, Alcor agent, was deployed as a cluster daemonset,
+which is running as container workload. All tenant level workload PODs, either containers or VMs will need Alcor's agent in
+order to have its network up. A pod can be scheduled to a node ONLY VM runtime ready. However, the pod cannot run because
+the network agent failed to setup the network for it.
+ 
+The following diagram illustrates the node status dependency, and cross workload type pod dependencies.
+
+![Design diagram](./Pod-dependency-cross-workload-types.PNG)
+
 This doc describes the proposal of the partial runtime readiness handling feature, including:
-1. Node readiness criteria with multiple runtime services on nodes
-2. Node runtime readiness state as part of the nodeStatus
+1. Node readiness criteria with multiple runtime services, with the Primary Runtime concept.
+2. Node runtime readiness state reported as part of the nodeStatus, nodeConditions.
 3. New scheduler predicate function for runtimeReadiness check
 
 ## Requirements, principles and terminologies
 ### Primary runtime of a cluster
-A primary runtime is the runtime service used for Kubernetes system daemon pods or pod with kube-system namespace. it is 
-usually a container runtime service given that the daemon runs as a process on the node. 
+A primary runtime is the runtime service used for Kubernetes system daemon pods, pod with kube-system namespace, or system
+workload that all other tenant workload pods rely on. In Arktos, the default primary runtime service is container workload 
+runtime service, because:
+
+1. Container workload is a simpler way to implement the daemonset pod than VM workload. 
+2. Currently Kubernetes daemonset pods are container based, default to container workload will make the migration to Arktos
+   more straightforward. 
 
 ### Runtime service state and state changes
 1. A Runtime service can be added or removed from a node
 2. A runtime service state can be at Ready, NotReady or Unknown; where Ready means both network and compute ready for 
-the runtime; NotReady means one of them or neither of them are ready in a particular runtime service. Unknown is the intial
+the runtime; NotReady means one of them or neither of them are ready in a particular runtime service. Unknown is the initial
```

Comment:
I don't suggest we use this: "where Ready means both network and compute ready for 
the runtime; NotReady means one of them or neither of them are ready in a particular runtime service.".  The runtime implementation is a black box to the node agent.  If it reports ready then it's ready, otherwise it's not ready.  If it's not ready, it could be many reasons (not only network and storage).  Node agent shouldn't care about the specific reasons.

Reply:
these condition checks in nodestatus is not added by the partial runtime feature.  i.e. kubelet's RuntimeServiceCondition function always checks for both compute and container network readiness for a given runtime service. This is an existing function. the description in the doc basically explains and emphasizes this current logic.

Reply:
I see. Thanks for the clarification.